sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1335,6 +1446,96 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
         }
     }
 
+    private CallAndOptionalResult<BlobDeleteResponse> deleteWithRestResponseIntern(String containerName,
+                                                                                   String blobName,
+                                                                                   String snapshot,
+                                                                                   Integer timeout,
+                                                                                   String version,
+                                                                                   String leaseId,
+                                                                                   DeleteSnapshotsOptionType deleteSnapshots,
+                                                                                   OffsetDateTime ifModifiedSince,
+                                                                                   OffsetDateTime ifUnmodifiedSince,
+                                                                                   String ifMatch,
+                                                                                   String ifNoneMatch,
+                                                                                   String requestId,
+                                                                                   Callback<BlobDeleteResponse> callback) {
+        DateTimeRfc1123 ifModifiedSinceConverted = ifModifiedSince == null ? null :
+            new DateTimeRfc1123(ifModifiedSince);
+        DateTimeRfc1123 ifUnmodifiedSinceConverted = ifUnmodifiedSince == null ? null :
+            new DateTimeRfc1123(ifUnmodifiedSince);
+
+        Call<Void> call = service.delete(containerName,
```

Comment:
This should be `Call<ResponseBody>` instead of  `Call<Void>`.

Reply:
The reason for "ResponseBody" is, we want later read the body if the delete fails i.e. if service returns non-202.